core/src/tasks/lookup_wallet.rs

```diff
@@ -177,243 +179,128 @@ impl LookupWalletTask {
 
     /// Find the wallet in the contract storage and create an opening for the wallet
     async fn find_wallet(&mut self) -> Result<(), LookupWalletTaskError> {
-        // Get the transaction that last updated the given wallet
-        let last_updated: TransactionHash = self
+        // Find the latest transaction updating the wallet, as indexed by the public share
+        // of the blinders
+        let mut blinder_csprng = PoseidonCSPRNG::new(self.blinder_seed);
+        let mut blinder_index = 0usize;
+        let curr_blinder = blinder_csprng.next().unwrap();
+        let curr_blinder_private_share = blinder_csprng.next().unwrap();
+        let mut curr_blinder_public = curr_blinder - curr_blinder_private_share;
+        let mut updating_tx = None;
+
+        while let Some(tx) = self
             .starknet_client
-            .get_wallet_last_updated(self.key_chain.public_keys.pk_view)
+            .get_public_blinder_tx(curr_blinder_public)
             .await
-            .map_err(|err| LookupWalletTaskError::Starknet(err.to_string()))?;
+            .map_err(|err| LookupWalletTaskError::Starknet(err.to_string()))?
+        {
+            updating_tx = Some(tx);
+            let curr_blinder = blinder_csprng.next().unwrap();
+            let curr_blinder_private_share = blinder_csprng.next().unwrap();
+            curr_blinder_public = curr_blinder - curr_blinder_private_share;
 
-        if last_updated == StarknetFieldElement::ZERO {
-            return Err(LookupWalletTaskError::NotFound(
-                ERR_WALLET_NOT_FOUND.to_string(),
-            ));
+            blinder_index += 1;
         }
```

Comment:
I mentioned this in your writeup as well, but this assumes that the owner of the given `sk_root` never experienced an MPC hangup, which would force them to do another `reblind` even though there was no update / settle transaction for the previous blinders.

You mentioned an approach to this being that we can check both if a blinder is indexed on-chain, and if it's been nullified, and consider the current wallet to be one where the public blinder is indexed on-chain but not nullified.

If this should be implemented after v0 that's fine, but it should be mentioned here as a `TODO`

Reply:
Added a todo.

I consider this as out of scope for v0, because the cause of the problem is not yet implemented either (i.e. we do not take any privacy preserving precautions after hangup yet). More of a privacy _optimization_ than a competion gating feature.